Mathlib/Tactic/MinImports.lean

```diff
@@ -65,12 +66,13 @@ def getSyntaxNodeKinds : Syntax → NameSet
   | .ident _ _ nm _ => NameSet.empty.insert nm
   | _ => {}
 
-/-- extracts the names of the declarations in `env` on which `decl` depends. -/
--- source:
--- https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Counting.20prerequisites.20of.20a.20theorem/near/425370265
-def getVisited (env : Environment) (decl : Name) : NameSet :=
-  let (_, { visited, .. }) := CollectAxioms.collect decl |>.run env |>.run {}
-  visited
+/-- Extracts the names of the declarations in `env` on which `decl` depends. -/
+def getVisited (decl : Name) : CommandElabM NameSet := do
+  unless (← hasConst decl) do
+    return {}
+  -- without resetting the state, the "unused tactics" linter gets confused?
+  let st ← get
+  liftCoreM decl.transitivelyUsedConstants <* set st
```

Comment:
@adomani Could you confirm that `getVisited` actually has to return the transitive closure of used constants for `#min_imports` to work, in contradiction to the `getAllDependencies` docstring?

Reply:
Yes, I think that this is correct.  The constants appearing here are what gets used at the end, there is no further closure taken.